### PR TITLE
[docs] - Graph-backed asset subsetting docs

### PR DESCRIPTION
**Pull request by clairelin135**
I recently added the functionality to subset graph-backed assets (i.e. selectively materialize a subset of assets outputted from the graph-backed asset).

Because this information is pretty dense reading, it might be overwhelming for a new user to see on the software-defined assets overview page. I broke out the graph-backed assets section into a new page and added the subsetting section at the bottom, marked as "advanced". 

Let me know how this looks!

[Preview Link](https://dagster-7wu5gqqec-elementl.vercel.app/concepts/assets/graph-backed-assets)

### CODE REVIEW

**Comment by sryza** [comment]
This is fantastic documentation. As you raised, this is a lot for users to digest when they're learning the basics of software-defined assets.  I think the idea you raised of making graph-backed assets their own page is the right idea.

**Comment by clairelin135** [comment]
@sryza glad you found it understandable! I broke out the graph-backed asset section into a new page (url is updated in PR description)

File: docs/content/concepts/assets/software-defined-assets.mdx
Comment (sryza) [inline docs/content/concepts/assets/software-defined-assets.mdx:214]: I think there's some overlap in these concepts that's worth pointing out here.  I would think of a graph-backed asset that outputs multiple assets as a kind of multi-asset.

The current wording might lead people to wonder about the difference between "A graph-backed asset can output multiple assets." and "A set of software-defined assets that are all updated by the same".

Comment (clairelin135) [reply docs/content/concepts/assets/software-defined-assets.mdx:214]: Changing it to this:

- [Multi-Assets](/concepts/assets/multi-assets): A set of software-defined assets that are all updated by the same [op](/concepts/ops-jobs-graphs/ops) or [graph](/concepts/ops-jobs-graphs/graphs).
- [Graph-Backed Assets](/concepts/assets/graph-backed-assets): A specific type of multi-asset whose computations are separated into multiple [ops](/concepts/ops-jobs-graphs/ops) that are combined to build a [graph](/concepts/ops-jobs-graphs/graphs).

Hopefully this is a little clearer

Comment (sryza) [reply docs/content/concepts/assets/software-defined-assets.mdx]: This makes it sound like all graph-backed assets are multi-assets, but I think it's more like a venn diagram where some graph-backed assets are multi-assets, some graph-backed assets aren't multi-assets, and some multi-assets aren't graph-backed assets.

File: docs/content/concepts/assets/graph-backed-assets.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/graph-backed-assets.mdx:20]: As I've been going through the Concept pages, I've been consolidating what was the **Overview** into the actual introduction section of the page. This eliminates what was a redundant/oddly-placed heading and makes the intro an actual introduction. Could we please do that here?

File: docs/content/concepts/assets/graph-backed-assets.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/graph-backed-assets.mdx:28]: I've also been adding horizontal rules before any H2-level headings to visually break up the page - can we please do that here?

```suggestion
---

## Defining graph-backed assets
```

File: docs/content/concepts/assets/graph-backed-assets.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/graph-backed-assets.mdx:133]: What do we hope including `Advanced` here will achieve?

File: docs/content/concepts/assets/graph-backed-assets.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/graph-backed-assets.mdx:139]: I've been working on making our diagram styles consistent ([example here](https://docs.dagster.io/concepts/io-management/io-managers)) - I'd be happy to make a version of this diagram that fits that style.

File: docs/content/concepts/assets/graph-backed-assets.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/graph-backed-assets.mdx:164]: This reads like there's a word missing here, so I guessed at the right one - feel free to toss this if it's not correct. I'm also suggesting adding a line break to visually break this up. I had to re-read this a few times to fully get it, as it felt really dense.

```suggestion
In the example, `foo` and `baz` produce outputs of `my_graph`. Subsequently, their outputs need to be yielded optionally. Because `foo` yields multiple outputs, we must structure our code to conditionally yield its outputs like in the code snippet above.

However, because `baz` only yields a singular output, Dagster will only run `baz` when its asset output `baz_asset` is selected. So, we don’t have to structure `baz` to return an optional output. Because `bar` does not yield any outputs that are returned from `my_graph`, its outputs do not have to be selectively returned.
```

File: docs/content/concepts/assets/software-defined-assets.mdx
Comment (erinkcochran87) [inline docs/content/concepts/assets/software-defined-assets.mdx:215]: These guys don't need to be capitalized:

```suggestion
- [Multi-assets](/concepts/assets/multi-assets): A set of software-defined assets that are all updated by the same [op](/concepts/ops-jobs-graphs/ops) or [graph](/concepts/ops-jobs-graphs/graphs).
- [Graph-backed assets](/concepts/assets/graph-backed-assets): A specific type of multi-asset whose computations are separated into multiple [ops](/concepts/ops-jobs-graphs/ops) that are combined to build a [graph](/concepts/ops-jobs-graphs/graphs).
```

**Review by erinkcochran87 — COMMENTED** [review]
This is looking great! I also agree breaking this out into its own page was the right call.

File: docs/content/concepts/assets/graph-backed-assets.mdx
Comment (clairelin135) [reply docs/content/concepts/assets/graph-backed-assets.mdx:133]: I think the content here is geared toward power users--it requires deeper preexisting knowledge about how ops are combined into graphs and the edge cases there.

My hope is that the "advanced" tag will signal to users (especially beginners) that they can skip this content unless if they are specifically looking for it and prepared to understand the nuances of this feature

File: docs/content/concepts/assets/graph-backed-assets.mdx
Comment (clairelin135) [reply docs/content/concepts/assets/graph-backed-assets.mdx:141]: That sounds great, thanks so much! I can add that in once you make it

Comment (erinkcochran87) [reply docs/content/concepts/assets/graph-backed-assets.mdx:141]: I'll knock it out later today!

File: docs/content/concepts/assets/software-defined-assets.mdx
Comment (clairelin135) [reply docs/content/concepts/assets/software-defined-assets.mdx]: Updated instead to this text:

- [Multi-assets](/concepts/assets/multi-assets): A set of software-defined assets that are all updated by the same [op](/concepts/ops-jobs-graphs/ops) or [graph](/concepts/ops-jobs-graphs/graphs).
- [Graph-backed assets](/concepts/assets/graph-backed-assets): An asset whose computations are separated into multiple [ops](/concepts/ops-jobs-graphs/ops) that are combined to build a [graph](/concepts/ops-jobs-graphs/graphs). If the graph outputs multiple assets, the graph-backed asset is a [multi-asset](/concepts/assets/multi-assets).

File: docs/content/concepts/assets/graph-backed-assets.mdx
Comment (clairelin135) [reply docs/content/concepts/assets/graph-backed-assets.mdx]: updated!

File: docs/content/concepts/assets/software-defined-assets.mdx
Comment (sryza) [reply docs/content/concepts/assets/software-defined-assets.mdx]: This looks great.

**Comment by sryza** [comment]
Looks good to me! Will leave it to @erinkcochran87 on the final approve.

**Review by erinkcochran87 — APPROVED** [review]
Looks good! There are some broken links to address before this can be merged though:

```


+ Array [
--
  | [2022-09-15T22:34:35Z]     +   Object {
  | [2022-09-15T22:34:35Z]     +     "deadLink": "concepts/assets/software-defined-assets#graph-backed-assets",
  | [2022-09-15T22:34:35Z]     +     "sourceFile": "/workdir/docs/content/concepts/ops-jobs-graphs/graphs.mdx",
  | [2022-09-15T22:34:35Z]     +   },
  | [2022-09-15T22:34:35Z]     +   Object {
  | [2022-09-15T22:34:35Z]     +     "deadLink": "concepts/assets/software-defined-assets#graph-backed-assets",
  | [2022-09-15T22:34:35Z]     +     "sourceFile": "/workdir/docs/content/concepts/ops-jobs-graphs/graphs.mdx",
  | [2022-09-15T22:34:35Z]     +   },
  | [2022-09-15T22:34:35Z]     +   Object {
  | [2022-09-15T22:34:35Z]     +     "deadLink": "software-defined-assets#a-basic-software-defined-asset",
  | [2022-09-15T22:34:35Z]     +     "sourceFile": "/workdir/docs/content/concepts/assets/graph-backed-assets.mdx",
  | [2022-09-15T22:34:35Z]     +   },
  | [2022-09-15T22:34:35Z]     +   Object {
  | [2022-09-15T22:34:35Z]     +     "deadLink": "concepts/assets/software-defined-assets#graph-backed-assets",
  | [2022-09-15T22:34:35Z]     +     "sourceFile": "/workdir/docs/content/concepts/assets/software-defined-assets.mdx",
  | [2022-09-15T22:34:35Z]     +   },
  | [2022-09-15T22:34:35Z]     + ]
```

**Pull request merged**
